### DISCUSSION TITLE
Adding DVB-I, DVB-HB and ATSC3 url’s to the channel list

### DISCUSSION THREAD

**Discussion by cablefarm** [PredatH0r/ChanSort#269] [Ideas]
There is a new standard out for finding and listing DVB-I channels, DVB-HB channels and ATSC3 NextGen channels. Generally, all it would need is to add the program source URL to the smart TV channel list.  And for the TV manufacturer to create the internal support hooks.  HbbTV and ATSC3 capable TV sets can already do most of that but you have to press the red or green button first and then select a program from a provided onscreen list. Some TV Apps also do those things.  No way to enter a new style DVB-I channel source URL or an ATSC3 channel source URL directly.  

It would be nice if we could enhance "ChanSort" to add Internet derived TV broadcast channels to the list so they can be selected from the TV remote by entering a channel number only.  Just like now it doesn't matter if the program comes from the analog tuner or a digital tuner, the TV viewer should be able to select an Internet delivered TV channel with the same ease.

**Comment by cablefarm** [comment]
Wa now have an ATSC 3 standard, also known as NextTV.  Mainly for the American market. It includes delivery of IP multicast based broadcast television from RF transmitters, and can also make 'on demand' content available over an Internet connetion.  Likewise to the HbbTV concept for the European market.  Time to have a good look how to process the service lists and give IP derived content a place in the program list so it can be called by pressing the channel number on the remote control.  Really, delivery of television content over IP connections is here to stay.  We better gear up for it.

**Comment by PredatH0r** [comment]
I'm not familiar with any of this and I can only add support for new formats when I have sample files.
So far I intentionally restrict ChanSort to modify only the order of existing channels in a list so I don't have to mess with all the internal data that a TV needs to tune in a specific channel. For most TV's it's simply impossible to reverse-engineer the data format or provide the actual data needed by the TV.
Channels that can be tuned simply by URL, like in m3u files, might be worth considering an exception.
